Marlin/pins_RIGIDBOT.h

```diff
@@ -0,0 +1,227 @@
+    #define BTN_EN2 59 // encoder
+    #define BTN_ENC 63 // enter button
+    #define SHIFT_OUT 40 // shift register
+    #define SHIFT_CLK 44 // shift register
+    #define SHIFT_LD 42 // shift register
+  #else
+    #define BTN_UP  37
+    #define BTN_DWN 35
+    #define BTN_LFT 33
+    #define BTN_RT  32
+    #define BTN_ENT 31
+    //#define BTN_EN1 37
+    //#define BTN_EN2 35
+    //#define BTN_ENC 31  //the click
+  #endif
+
+  #ifdef G3D_PANEL
+    #define SDCARDDETECT 49
+  #else
+    #define SDCARDDETECT -1  // Ramps does not use this port
+  #endif
+
+#endif
+
+
+// SPI for Max6675 Thermocouple
+#ifndef SDSUPPORT
+  #define MAX6675_SS       53 // Do not use pin 53 if there is even the remote possibility of using Display/SD card
+#else
+  #define MAX6675_SS       49 // Do not use pin 49 as this is tied to the switch inside the SD card socket to detect if there is an SD card present
+#endif
+
+#ifndef SDSUPPORT
+  // these pins are defined in the SD library if building with SD support
+  #define SCK_PIN          52
+  #define MISO_PIN         50
+  #define MOSI_PIN         51
+#endif
+
+
+
+/****************************************************************************************
+* RigidBot Arduino Mega pin assignment
+*
+****************************************************************************************/
+//#define X_STEP_PIN         54
+//#define X_DIR_PIN          55
+//#define X_ENABLE_PIN       38
+//#define X_MIN_PIN           3
+//#define X_MAX_PIN           2
+
+//#define Y_STEP_PIN         60
+//#define Y_DIR_PIN          61
+//#define Y_ENABLE_PIN       56
+//#define Y_MIN_PIN          14
+//#define Y_MAX_PIN          15
+ 
+//#define Z_STEP_PIN         46
+//#define Z_DIR_PIN          48
+//#define Z_ENABLE_PIN       62
+//#define Z_MIN_PIN          18
+//#define Z_MAX_PIN          19
+//#define Z_PROBE_PIN        -1
+
+//#define Y2_STEP_PIN        36  // Not supported by Rigidbot
+//#define Y2_DIR_PIN         34  // Not supported by Rigidbot
+//#define Y2_ENABLE_PIN      30  // Not supported by Rigidbot
+
+//#define Z2_STEP_PIN        36
+//#define Z2_DIR_PIN         34
+//#define Z2_ENABLE_PIN      30
+
+#undef E0_STEP_PIN
+#undef E0_DIR_PIN
+#undef E0_ENABLE_PIN
+
+#define E0_STEP_PIN        36
+#define E0_DIR_PIN         34
+#define E0_ENABLE_PIN      30
+
+#undef E1_STEP_PIN
+#undef E1_DIR_PIN
+#undef E1_ENABLE_PIN
+
+#define E1_STEP_PIN        26
+#define E1_DIR_PIN         28
+#define E1_ENABLE_PIN      24
+
+//#define SDPOWER            -1
+//#define SDSS               53
+//#define LED_PIN            13
+
+#undef  FAN_PIN
+#define FAN_PIN             8 //11 // IO pin. Buffer needed
+
+#undef  PS_ON_PIN
+#define PS_ON_PIN          -1
+
+#undef  HEATER_BED_PIN
+#define HEATER_BED_PIN     10   // BED
+
+#undef  TEMP_BED_PIN
+#define TEMP_BED_PIN       15   // ANALOG NUMBERING
+ 
+
+#define STEPPER_RESET_PIN 41 // #define STEPPER_RESET_FIX in Configuration.h to use this during boot
```

Comment:
I'm pretty certain this file can be a lot more concise, unless the Rigidbot board is very unusual. Look at the other `pins_SOME_BOARD.h` files that are variants on `RAMPS_13` and see if yours can't be similarly compact.